File: Daily/16_03_2594/2594.java
class Solution {
    public long repairCars(int[] ranks, int cars) {
      Array.sort(ranks);
      long left = 0;
      long right = (long) ranks[0] * cars * cars;
      long ans = right;

      while(left <= right) {
        long mid = left + (right - left) / 2;
        long totalCarsRepaired = 0;
        for (int rank : ranks) {
            totalCarsRepaired += (long) Math.sqrt((double) mid / rank);
        }

        if (totalCarsRepaired >= cars) {
            ans = mid;
            right = mid - 1;
        } else {
            left = mid + 1;
        }
    }
        return ans;
      
}}

File: Daily/MARCH/16_03_2594/2594.js
/**
 * @param {number[]} ranks
 * @param {number} cars
 * @return {number}
 */
var repairCars = function (ranks, cars) {
  ranks.sort((a, b) => a - b);
  let left = 0;
  let right = ranks[0] * cars * cars;
  let ans = right;

  while (left <= right) {
    const mid = Math.floor((left + right) / 2);
    let count = 0;
    for (const rank of ranks) {
      count += Math.floor(Math.sqrt(mid / rank));
    }

    if (count >= cars) {
      ans = mid;
      right = mid - 1;
    } else {
      left = mid + 1;
    }
  }
  return ans;
};
